scrapers/scraper.py

```diff
@@ -0,0 +1,267 @@
+            for line in lines:
+                if line.startswith('commit'):
+                    commitData["sha"] = line[len("commit "):]
+                if line.startswith('Author'):
+                    commitData["author"] = line.split(": ")[1].split(" <")[0]
+                if line.startswith('Date'):
+                    commitData["date"] = line[len("date:"):].strip()
+
+            if any(commitData["sha"] == cm["sha"] for cm in history):
+                continue
+
+            dixFile = "https://raw.githubusercontent.com/apertium/apertium-{0}/{1}/apertium-{0}.{0}.dix".format(pairName, commitData["sha"])
+            stems = countDixStems(dixFile, True)
+            if stems == -1:
+                dixName = iso3to2[pairList[0]] + "-" + iso3to2[pairList[1]]
+                dixFile = "https://raw.githubusercontent.com/apertium/apertium-{0}/{2}/apertium-{1}.{1}.dix".format(pairName, dixName, commitData["sha"])
+                stems = countDixStems(dixFile, True)
+                if stems == -1:
+                    continue
+
+            commitData["stems"] = stems["stems"]
+            history.append(commitData)
+
+        langPackages.append({"name": pairName, "history": history})
+    return langPackages
+
+def monoData(packages, updatemailmap):
+    """Returns data for all monolingual dictionaries: state, stems, location and contributors"""
+    data = []
+    for package in packages:
+        if not(re.match(r"apertium-\w+$", package["name"]) and package["name"][len("apertium-"):] in languages):
+            continue
+
+        language = package["name"][len("apertium-"):]
+        stats = json.loads(requests.get("https://apertium.projectjj.com/stats-service/apertium-{}/Lexc".format(language)).text)["stats"]
+        for statistic in stats:
+            if statistic["stat_kind"] == "Stems":
+                stems = statistic["value"]
+                break
+        for topic in package["topics"]:
+            if topic[len("apertium-"):] in langLocations:
+                location = topic[len("apertium-"):]
+                break
+
+        subprocess.call(['git', 'clone', 'https://github.com/apertium/apertium-{}'.format(language)], stderr=subprocess.DEVNULL)
+        shutil.copyfile("./.mailmap", "./apertium-{}/.mailmap".format(language))
+        lines = subprocess.check_output(['git', 'log', '--format=%aN', '--follow', 'apertium-{0}.{0}.lexc'.format(language)], cwd='./apertium-{}/'.format(language)).decode('utf-8')
+
+        if updatemailmap:
+            emails = subprocess.check_output(['git', 'log', '--format=<%aE>', '--follow', 'apertium-{0}.{0}.lexc'.format(language)], cwd='./apertium-{}/'.format(language)).decode('utf-8').split("\n")
+            mailmap = open("./.mailmap", "r", encoding="utf-8").read()
+            for email in emails:
+                if not email in mailmap:
+                    print(email)
+
+        shutil.rmtree("./apertium-{}/".format(language), onerror=del_rw)
```

Comment:
Were you actually running into situations where you needed to mess with permission bits?

Reply:
Yes, I kept getting Access Denied

Reply:
Are you running the script as yourself or root? Maybe you're mixing WSL and Windows somehow..?

Reply:
I don't use WSL, but rather have installed pretty much everything I need. I've tried running git bash as admin, but that doesn't work either. I don't rly understand why, but his workaround fixes it